torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1277,6 +1277,19 @@ def statically_known_true(x: Union[bool, SymBool]) -> bool:
     return x
 
 
+# When a or b is evaluated, a is evaluated eagerly first then b. This causes
+# a data dependent error for an expression “if u0==1 or True”. or over guarding for
+# “if s0==1 or True”.
+
+# On the other hand, when we use operator.or_, then dynamo will generate
+# a sympy expression Sympy.Or(u0==1, True) without evaluating the args first.
+
+# When the whole expression is passed to evaluation in that case, we do not throw a
+# data dependent error or guard because we can statically know the result is True
+# before unpacking the symbols.
+sym_or = operator.or_
```

Comment:
given that @pian is introducing sym_or in https://github.com/pytorch/pytorch/pull/150456#event-17082477541, does it make sense to rebase on top of his PR?